utils/src/main/scala/com/normation/utils/Control.scala

```diff
@@ -74,10 +74,15 @@ object Control {
   def sequencePar[U,T](seq:ParSeq[U])(f:U => Box[T]) : Box[Seq[T]] = {
     val buf = scala.collection.mutable.Buffer[T]()
     @volatile var stop = Option.empty[EmptyBox]
-    seq.foreach { u => if(stop.isEmpty) f(u) match {
-      case Full(x) => buf += x
-      case e:EmptyBox => stop = Some(e)
-    } }
+    seq.foreach { u =>
+      if(stop.isEmpty) { f(u) match {
+        case Full(x) =>
+          buf.synchronized{
```

Comment:
We looked with @fanf for a "concurrent"  collection, but we could find one that was ok.

I would have liked a fold into an immutable Seq, but i think there would ha ve been problems too ...